h5netcdf/core.py

```diff
@@ -125,28 +132,61 @@ def shuffle(self):
 
 NOT_A_VARIABLE = b'This is a netCDF dimension but not a netCDF variable.'
 
+class lazy_objects(Mapping):
+    def __init__(self, object_cls):
+        self._objects = list()
```

Comment:
I might call this "unloaded_object" for clarity. Also, make it a set so contains checks are fast.